src/components/Airdrop/components/Airdrop/Airdrop.tsx

```diff
@@ -9,34 +8,32 @@ import { t } from "i18next";
 import { useCallback, useEffect, useState } from "react";
 import { useSelector } from "react-redux";
 import { RootState } from "reducers";
-import { isEthereumProvider, normalizeAddress } from "../../../../utils";
-import TokenAirdrop from "../TokenAirdrop/TokenAirdrop";
+import { normalizeAddress } from "../../../../utils";
+//import TokenAirdrop from "../TokenAirdrop/TokenAirdrop";
 import NFTAirdrop from "../NFTAirdop/NFTAirdrop";
+import { useParams } from "react-router-dom";
+import { useActions } from "actions/contractsActions";
 import "./index.scss";
-import { useSearchParams } from "react-router-dom";
 
 export default function Airdrop() {
-  const [userInput, setUserInput] = useState("");
-  const [pendingWallet, setPendingWallet] = useState(false);
+  const { hasClaimed, isRedeemed } = useActions();
+  const { walletAddress } = useParams();
+  const [userInput, setUserInput] = useState(walletAddress);
   const nftET = useSelector((state: RootState) => state.dataReducer.airdrop?.nft);
   const tokenET = useSelector((state: RootState) => state.dataReducer.airdrop?.token);
   const [nftEligibilityStatus, setNFTEligibilityStatus] = useState(EligibilityStatus.UNKNOWN);
```

Comment:
i think this state is not need as it is calculated once we read from the contract

Reply:
@leviyehonatan ok I'm sure this code can be improved but that's for a separate PR. We first need to merge this.